cmd/remediation-aggregator/main.go

```diff
@@ -0,0 +1,426 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+	"reflect"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sort"
+	"strings"
+	"time"
+)
+
+const (
+	configMapRemediationsProcessed = "compliance-remediations/processed"
+	configMapCompressed            = "openscap-scan-result/compressed"
+	nodeRolePrefix                 = "node-role.kubernetes.io/"
+)
+
+type aggregatorConfig struct {
+	Content   string
+	ScanName  string
+	Namespace string
+}
+
+type complianceCrClient struct {
+	client runtimeclient.Client
+	scheme *runtime.Scheme
+}
+
+func defineFlags(cmd *cobra.Command) {
+	cmd.Flags().String("content", "", "The path to the OpenScap content")
+	cmd.Flags().String("scan", "", "The compliance scan that owns the configMap objects.")
+	cmd.Flags().String("namespace", "openshift-compliance", "Running pod namespace.")
+}
+
+func parseConfig(cmd *cobra.Command) *aggregatorConfig {
+	var conf aggregatorConfig
+	conf.Content = getValidStringArg(cmd, "content")
+	conf.ScanName = getValidStringArg(cmd, "scan")
+	conf.Namespace = getValidStringArg(cmd, "namespace")
+	return &conf
+}
+
+func getValidStringArg(cmd *cobra.Command, name string) string {
+	val, _ := cmd.Flags().GetString(name)
+	if val == "" {
+		fmt.Fprintf(os.Stderr, "The command line argument '%s' is mandatory.\n", name)
+		os.Exit(1)
+	}
+	return val
+}
+
+func createCrClient(config *rest.Config) (*complianceCrClient, error) {
+	scheme := runtime.NewScheme()
+
+	v1.AddToScheme(scheme)
+	mcfgv1.AddToScheme(scheme)
+
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&complianceoperatorv1alpha1.ComplianceRemediation{})
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&complianceoperatorv1alpha1.ComplianceScan{})
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&metav1.CreateOptions{})
+
+	client, err := runtimeclient.New(config, runtimeclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &complianceCrClient{
+		client: client,
+		scheme: scheme,
+	}, nil
+}
+
+func getScanRoleLabel(nodeSelector map[string]string) string {
+	if nodeSelector == nil {
+		return ""
+	}
+
+	// FIXME: should we protect against multiple labels and return
+	// an empty string if there are multiple?
+	for k := range nodeSelector {
+		if strings.HasPrefix(k, nodeRolePrefix) {
+			return strings.TrimPrefix(k, nodeRolePrefix)
+		}
+	}
+
+	return ""
+}
+
+func getScanConfigMaps(clientset *kubernetes.Clientset, scan, namespace string) ([]v1.ConfigMap, error) {
+	var cMapList *v1.ConfigMapList
+	var err error
+
+	// Look for configMap with this scan label
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
We probably don't need this Poll anymore, do we?

Reply:
I guess not since one of your last patches, fixed locally